src/components/Card/Card.js

```diff
@@ -15,19 +17,21 @@ export default function Cards({ activities, categories, deleteActivity }) {
             : sumHoursByCategory(activities, category)
         }
       />
-      <ActivitySection>
-        {activities
-          .filter(activity => activity.category === category)
-          .map(activity => (
-            <Activity
-              activity={activity}
-              key={uid(activity)}
-              deleteActivity={deleteActivity}
-            />
-          ))}
-      </ActivitySection>
+      {on && (
+        <ActivityStyled>
+          {activities
+            .filter(activity => activity.category === category)
+            .map(activity => (
+              <Activity
+                activity={activity}
+                key={uid(32)}
+                deleteActivity={deleteActivity}
+              />
+            ))}
+        </ActivityStyled>
+      )}
     </CardStyled>
-  ))
+  )
 
   function sumHoursByCategory(activities, category) {
     const sum = activities.reduce(
```

Comment:
No need to manually assign to sum in reducer below, simply go with `activityItem.category === category ? sum + activityItem.hours : sum`. The accumulator automatically gets assign the return value of the reducer function.
Nice way to avoid the filtereing before summing!